tests/impls/detect_image_objects/test_centernetvisdrone.py

```diff
@@ -198,3 +205,26 @@ def mock_model_forward(img_tensors: "torch.Tensor") -> Dict:  # noqa: F821
         "wh": test_wh,
         "reg": test_reg,
     }
+
+
+@pytest.mark.skipif(not CenterNetVisdrone.is_usable(),
+                    reason="CenterNetVisdrone is not usable.")
+def test_gather_feat_mps() -> None:
+    """
+    Check that alternative implementation for torch.gather on MPS
+    matches torch.gather.
+    """
+    torch.manual_seed(42)
+
+    batch_size = 2
+    num_features = 76800
+    selected_features = 256
+    dim = 2
+
+    feat = torch.randn(batch_size, num_features, dim, device="cpu")
+    ind = torch.randint(0, num_features, (batch_size, selected_features, dim), device="cpu")
+
+    feat_cpu = _gather(feat, ind, "cpu")
+    feat_mps = _gather(feat, ind, "mps")
```

Comment:
Able to test the Mac implementation by passing the device type manually (rather than having `torch` infer from the current hardware)